OpenDreamShared/Compiler/DM/DMParser.cs

```diff
@@ -1579,7 +1579,18 @@ public DMASTProcStatementThrow Throw()
 
             if (value != null && Check(TokenType.DM_In)) {
                 Whitespace();
-                DMASTExpression list = ExpressionIn();
+                DMASTExpression list = Expression();
+
+                Whitespace();
+                if (Check(TokenType.DM_To))
+                {
+                    Whitespace();
+                    DMASTExpression endRange = Expression();
+                    if (endRange is not null)
+                    {
+                        return new DMASTExpressionInRange(value, list, endRange);
+                    }
+                }
 
                 return new DMASTExpressionIn(value, list);
```

Comment:
From what I can tell `A in B` and `A in B to C` have the same precedence and take priority over ternaries. Move `ExpressionIn()` between `ExpressionAssign()` and `ExpressionTernary()`.